frontend/src/components/TwoLevelDropdown.tsx

```diff
@@ -0,0 +1,295 @@
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    maxWidth: '30rem',
+    overflow: 'hidden',
+    verticalAlign: 'top',
+  },
+  dropdownSubmenu: {
+    left: '100%',
+    // Offsets the dropdown menu vertical padding (5px) and border (1px).
+    top: '-6px',
+  },
+  inlineContainer: {
+    display: 'inline-block',
+  },
+  relativeContainer: {
+    position: 'relative',
+  },
+});
+
+const ChevronRightSmallIcon = () => <ChevronRightIcon className={classes(css.defaultFont)} />;
+
+export interface SelectedItem {
+  itemName: string;
+  subItemName: string;
+  subItemSecondaryName?: string;
+}
+
+interface DropdownSubItem {
+  name: string;
+  secondaryName?: string;
+}
+
+export interface DropdownItem {
+  name: string;
+  subItems: DropdownSubItem[];
+}
+
+interface DropdownButtonProps {
+  dropdownRef: Ref<HTMLDivElement>;
+  toggleDropdown: (e: React.MouseEvent) => void;
+  selectedItem: SelectedItem;
+  title: string;
+}
+
+function DropdownButton(props: DropdownButtonProps) {
+  const { dropdownRef, toggleDropdown, selectedItem, title } = props;
+
+  const dropdownTooltip: string =
+    selectedItem.itemName && selectedItem.subItemName
+      ? `${selectedItem.itemName} > ${selectedItem.subItemName}` +
+        (selectedItem.subItemSecondaryName ? ` > ${selectedItem.subItemSecondaryName}` : '')
+      : title;
+  return (
+    <div className={classes(css.inlineContainer)} ref={dropdownRef}>
+      <Button onClick={toggleDropdown} className={classes(css.dropdown, css.textContainer)}>
+        <Tooltip title={dropdownTooltip} enterDelay={300} placement='top-start'>
+          <span className={classes(css.textContainer)}>
+            {selectedItem.itemName && selectedItem.subItemName ? (
+              <>
+                {selectedItem.itemName}
+                <ChevronRightSmallIcon />
+                {selectedItem.subItemName}
+                {selectedItem.subItemSecondaryName && (
+                  <>
+                    <ChevronRightSmallIcon />
+                    {selectedItem.subItemSecondaryName}
+                  </>
+                )}
+              </>
+            ) : (
+              title
+            )}
+          </span>
+        </Tooltip>
+        <ArrowDropDownIcon />
+      </Button>
+    </div>
+  );
+}
+
+interface DropdownSubMenuProps {
+  subDropdownActive: number;
+  setSubDropdownActive: (subDropdownActive: number) => void;
+  subDropdownIndex: number;
+  item: DropdownItem;
+  setSelectedItem: (selectedItem: SelectedItem) => void;
+  setDropdownActive: (dropdownActive: boolean) => void;
+}
+
+function DropdownSubMenu(props: DropdownSubMenuProps) {
+  const {
+    subDropdownActive,
+    setSubDropdownActive,
+    subDropdownIndex,
+    item,
+    setSelectedItem,
+    setDropdownActive,
+  } = props;
+
+  if (item.subItems.length === 0 || subDropdownActive !== subDropdownIndex) {
+    return <></>;
+  }
+
+  return (
+    <ul className={classes(css.dropdownSubmenu, css.dropdownMenu)}>
```

Comment:
Do we need to sort / group `subItems` here so `subItems` with same `subItemsName` but different `SecondaryName` are grouped together in the rendered list? Or such data has been sorted on higher component?

Reply:
Same as the answer in the discussion https://github.com/kubeflow/pipelines/pull/7933#discussion_r906727983, I plan to implement the sorting on a higher component (`CompareV2.tsx` in https://github.com/kubeflow/pipelines/pull/7943).